### PR TITLE
Network: fabric geneve interfaces

### PR DESCRIPTION
**Pull request by cheina97**
# Description

This PR adds the creation of the geneve interfaces inside the deamonset, and uses the GeneveTunnel resources Ids to setup the geneve tunnels.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -76,7 +76,7 @@ rbacs: controller-gen
...
 	$(CONTROLLER_GEN) paths="{./pkg/gateway/...,./cmd/gateway/...,./pkg/firewall/...,./pkg/route/...}" rbac:roleName=liqo-gateway output:rbac:stdout | awk -v RS="---\n" 'NR>1{f="./deployments/liqo/files/liqo-gateway-" $$4 ".yaml";printf "%s",$$0 > f; close(f)}' && $(SED_COMMAND) deployments/liqo/files/liqo-gateway-ClusterRole.yaml
-	$(CONTROLLER_GEN) paths="{./cmd/fabric/...,./pkg/firewall/...,./pkg/route/...}" rbac:roleName=liqo-fabric output:rbac:stdout | awk -v RS="---\n" 'NR>1{f="./deployments/liqo/files/liqo-fabric-" $$4 ".yaml";printf "%s",$$0 > f; close(f)}' && $(SED_COMMAND) deployments/liqo/files/liqo-fabric-ClusterRole.yaml
+	$(CONTROLLER_GEN) paths="{./cmd/fabric/...,./pkg/firewall/...,./pkg/route/...,./pkg/fabric}" rbac:roleName=liqo-fabric output:rbac:stdout | awk -v RS="---\n" 'NR>1{f="./deployments/liqo/files/liqo-fabric-" $$4 ".yaml";printf "%s",$$0 > f; close(f)}' && $(SED_COMMAND) deployments/liqo/files/liqo-fabric-ClusterRole.yaml
```
Comment (aleoli) [inline Makefile:79]:
```suggestion
	$(CONTROLLER_GEN) paths="{./cmd/fabric/...,./pkg/firewall/...,./pkg/route/...,./pkg/fabric/...}" rbac:roleName=liqo-fabric output:rbac:stdout | awk -v RS="---\n" 'NR>1{f="./deployments/liqo/files/liqo-fabric-" $$4 ".yaml";printf "%s",$$0 > f; close(f)}' && $(SED_COMMAND) deployments/liqo/files/liqo-fabric-ClusterRole.yaml
```

File: pkg/gateway/fabric/geneve/internalnode_controller.go [hunk 2]
```diff
@@ -65,10 +66,23 @@ func (r *InternalNodeReconciler) Reconcile(ctx context.Context, req ctrl.Request
 
 	klog.V(4).Infof("Reconciling internalnode %s", req.String())
 
-	if err := EnsureGeneveInterfacePresence(internalnode, r.Options); err != nil {
+	// The internal fabric has the same name of the gateway resource.
```
Comment (aleoli) [inline pkg/gateway/fabric/geneve/internalnode_controller.go:69]: It seems quite unsafe, can you at least define a variable internalNodeName with this comment, and then use that variable?

```suggestion
	// The internal fabric has the same name of the gateway resource.
    internalNodeName := r.Options.GwOptions.Name
    id, err := geneve.GetGeneveTunnelID(ctx, r.Client, internalNodeName, internalnode.Name)
```

File: pkg/utils/geneve/k8s.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	}
+	if len(list.Items) == 0 {
+		return 0, fmt.Errorf("no geneve tunnel found for internalfabric %s and internalnode %s",
+			internalFabricName, internalNodeName)
+	}
+	if len(list.Items) > 1 {
+		return 0, fmt.Errorf("multiple geneve tunnels found for internalfabric %s and internalnode %s",
+			internalFabricName, internalNodeName)
+	}
+	return list.Items[0].Spec.ID + 1000, nil
```
Comment (aleoli) [inline pkg/utils/geneve/k8s.go:48]: Can we use a switch for that?

**Review by aleoli — APPROVED** [review]

File: pkg/gateway/fabric/geneve/internalnode_controller.go (hunk 2 above)
Comment (fra98) [reply pkg/gateway/fabric/geneve/internalnode_controller.go:69]: We could use the generic function `gateway.GenerateResourceName()`` that we used for other resources associated with the gateway (and also use it when creating the internalfabrics in the server_controller and client_controller)

Comment (cheina97) [reply pkg/gateway/fabric/geneve/internalnode_controller.go:69]: @fra98 I agree with you.

**Pull request merged**